### PR TITLE
Generic integration test bundle + 2 simple Integration tests

### PR DESCRIPTION
**Pull request by f2par0**
Purpose : 
Provide a generic bundle containing code that can be used by most of components to test.

- camel-integration-test contains the configuration (users.properties + config() method)  for the karaf test containers,  utility methods, and an abstract method `createRouteBuilder` that must be overridden by tests classes.
- camel-file and camel-seda are 2 simple tests inheriting the base class 
- tests are minimal, just to make sure the bundle is installed and the route behave correctly, the goal is not to retest throughly the component which is done by camel integration tests.

Next : for more complex components which require external process (a db, server ...), goal is to use test containers but this doesn't work yet.

### CODE REVIEW

File: components/camel-file/pom.xml [hunk 1]
```diff
@@ -40,7 +40,17 @@
...
+        <dependencies>
+            <dependency>
+                <groupId>org.apache.karaf</groupId>
+                <artifactId>karaf-bom</artifactId>
+                <version>${karaf.version}</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
```
Comment (essobedo) [inline components/camel-file/pom.xml]: Don't seem to be used, I guess we can remove it

File: pom.xml [hunk 2]
```diff
@@ -182,6 +184,18 @@
...
+                <artifactId>org.apache.servicemix.bundles.hamcrest</artifactId>
+                <version>${harmcrest.version}</version>
+                <scope>runtime</scope>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.geronimo.specs</groupId>
+                <artifactId>geronimo-atinject_1.0_spec</artifactId>
+                <version>${geronimo-atinject.version}</version>
+                <scope>test</scope>
+            </dependency>
```
Comment (essobedo) [inline pom.xml:198]: I would move it into tests/components/pom.xml, but I understand that it is more or less controversial so let's see what JB thinks

File: tests/camel-integration-test/pom.xml
Comment (essobedo) [inline tests/camel-integration-test/pom.xml:1]: Warning the copyright header is missing

File: tests/camel-integration-test/src/main/java/org/apache/karaf/camel/itests/AbstractCamelComponent.java
Comment (essobedo) [inline tests/camel-integration-test/src/main/java/org/apache/karaf/camel/itests/AbstractCamelComponent.java:1]: Warning the copyright header is missing

File: tests/components/camel-file/pom.xml
Comment (essobedo) [inline tests/components/camel-file/pom.xml:1]: Warning the copyright header is missing

File: tests/components/camel-seda/pom.xml
Comment (essobedo) [inline tests/components/camel-seda/pom.xml:1]: Warning the copyright header is missing

**Comment by jbonofre** [comment]
@essobedo @f2par0 can you please wait my review before merging ? I identified a couple of things that should be improved/changed. I will do that in a separate PR.

**Comment by essobedo** [comment]
@jbonofre my bad, let's revert it

**Comment by jbonofre** [comment]
@essobedo no need to revert, let's keep it. It's a working branch. I will do a separate PR with some improvements.

**Comment by essobedo** [comment]
@jbonofre ok 👍

**Pull request merged**